python_modules/dagster/dagster/_core/definitions/asset_condition/asset_condition.py

```diff
@@ -229,7 +181,8 @@ def legacy_num_discarded(self) -> int:
 
 
 @whitelist_for_serdes
-class AssetConditionEvaluationState(NamedTuple):
+@dataclass(frozen=True)
```

Comment:
has to be a dataclass because it's recursive

Reply:
does this not work? 
https://docs.pydantic.dev/1.10/usage/postponed_annotations/#self-referencing-models